include/lsst/afw/geom/SkyWcs.h

```diff
@@ -0,0 +1,205 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_AFW_GEOM_SKYWCS_H
+#define LSST_AFW_GEOM_SKYWCS_H
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "Eigen/Core"
+#include "astshim.h"
+#include "ndarray.h"
+
+#include "lsst/afw/geom/Angle.h"
+#include "lsst/afw/geom/Point.h"
+#include "lsst/afw/geom/SpherePoint.h"
+#include "lsst/afw/geom/Endpoint.h"
+#include "lsst/afw/geom/Transform.h"
+#include "lsst/daf/base/PropertyList.h"
+
+namespace lsst {
+namespace afw {
+namespace geom {
+
+/**
+Make a WCS CD matrix
+
+@param[in] scale  Pixel scale as an angle on sky/pixels
+@param[in] orientation  Position angle of focal plane +Y, measured from N through E.
+                        At 0 degrees, +Y is along N and +X is along E/W if flipX false/true
+                        At 90 degrees, +Y is along E and +X is along S/N if flipX false/true
+@param[in] flipX  Fip x axis (see orientation for details)?
+
+@return the CD matrix, where element (i-1, j-1) corresponds to FITS keyword CDi_j
+                        and i, j have range [1, 2]
+
+*/
+Eigen::Matrix2d makeCdMatrix(Angle const &scale, Angle const &orientation = 0 * degrees, bool flipX = false);
+
+/**
+A WCS that transform pixels to ICRS RA/Dec using the LSST standard for pixels
+
+The LSST standard is: 0,0 is the lower left pixel of the parent image
+(unlike FITS, which uses 1,1 as the lower left pixel of the subimage).
+
+The forward direction transforms from pixels (actual, if known, else nominal) to ICRS RA, Dec.
+
+@anchor skywcs_frames The contained ast::FrameSet must have the following frames.
+All are of class ast::Frame except the sky frame. The domain is listed first, if set:
+- ACTUAL_PIXEL0 (optional): actual pixels in LSST convention: 0,0 is lower left corner of parent image.
+    The 0 is to remind users that this frame is 0-based.
+- PIXEL0: nominal pixels in LSST convention: 0,0 is lower left corner of parent image
+- GRID: nominal pixels in FITS convention: 1,1 is lower left corner of subimage
+- IWC: intermediate world coordinates (the FITS WCS concept)
+- An ast::SkyFrame with System=ICRS and standard axis order RA, Dec.
+
+If ACTUAL_PIXEL0 is present then it will be the base frame, else PIXEL0 will be the base frame.
+The SkyFrame is always the current frame.
```

Comment:
Good point. I put in a bit more text. I am hoping it will suffice, and we can find a better place for if more is needed.